components/create_comment/create_comment.jsx

```diff
@@ -334,9 +344,17 @@ class CreateComment extends React.PureComponent {
         e.preventDefault();
 
         const {draft} = this.state;
-        let message = '';
+        let message = draft.message;
+
         if (isGitHubCodeBlock(table.className)) {
-            message = '```\n' + getPlainText(e.clipboardData) + '\n```';
+            const codeBlock = '```\n' + getPlainText(e.clipboardData) + '\n```';
+            const {firstPiece, lastPiece} = splitMessageBasedOnCaretPosition(this.state.caretPosition, message);
+
+            // check whether the first piece of the message is empty when cursor is placed at beginning of message and avoid adding an empty string at the beginning of the message
+            message = firstPiece === '' ? `${codeBlock}\n${lastPiece}` : `${firstPiece}\n${codeBlock}\n${lastPiece}`;
+
+            const newCaretPosition = firstPiece === '' ? `${codeBlock}`.length : `${firstPiece} ${codeBlock}`.length;
```

Comment:
You can simplify these lines calculating if you need the extra LF first. Also added a check for LF after the codeblock to prevent adding an innecesary one (not that most people would notice).
```suggestion
            const {firstPiece, lastPiece} = splitMessageBasedOnCaretPosition(this.state.caretPosition, message);
            const requireStartLF = firstPiece === '' ? '' : '\n';
            const requireEndLF = lastPiece === '' ? '' : '\n';
            const codeBlock = requireStartLF + '```\n' + getPlainText(e.clipboardData) + '\n```' + requireEndLF;

            // check whether the first piece of the message is empty when cursor is placed at beginning of message and avoid adding an empty string at the beginning of the message
            message = `${firstPiece}${codeBlock}${lastPiece}`;

            const newCaretPosition = this.state.caretPosition + codeBlock.length;
```

Reply:
Makes sense. TBH I pretty much copied this from a similar method.

Reply:
That was a good suggestion 👍. Thank you